sympy/functions/special/hyper.py

```diff
@@ -191,6 +195,22 @@ class hyper(TupleParametersBase):
 
     def __new__(cls, ap, bq, z, **kwargs):
         # TODO should we check convergence conditions?
+        ca = Counter(Tuple(*ap))
+        cb = Counter(Tuple(*bq))
+        for i in ca:
+            if i in cb and i.is_nonnegative:
+                if ca[i] > cb[i]:
```

Comment:
Will this raise for symbolic expressions?

Reply:
I thought the same thing when reviewing, but no: a counter's values are ints and Counter is happy to collect on symbols:

```python
>>> Counter(var('x x y'))
Counter({x: 2, y: 1})
```

Reply:
Is this like:
```python
In [64]: c1 = Counter([1, 2, 3, 1])

In [65]: c2 = Counter([1, 3, 3, 1])

In [66]: c1 - (c1 & c2)
Out[66]: Counter({2: 1})

In [67]: c2 - (c1 & c2)
Out[67]: Counter({3: 1})
```

Reply:
That's a nice feature, more set like than I knew. Code is being updated.